### ISSUE TITLE
rustfmt instructions

### ISSUE THREAD

**Issue by sayrer** [ctz/rustls#582]
I tried running `rustfmt` on the whole crate, and it came up with like 30 changes. What's the best way to run this in advance of a PR? Should I be running a nightly `rustfmt`?

**Comment by djc** [comment]
Yes, like this: https://github.com/ctz/rustls/blob/main/.github/workflows/build.yml#L170.

I was thinking I could do an `admin/fmt` script that takes care of building (similar to what we have for bogo) and running it.

**Comment by sayrer** [comment]
Hmm, I still get a lot of changes this way:

```
% rustup run nightly cargo version      
cargo 1.52.0-nightly (90691f2bf 2021-03-16)
% rustup run nightly cargo-fmt --version
rustfmt 1.4.36-nightly (7de6968e 2021-02-07)
% rustup run nightly cargo-fmt 

...[lots of changes]...
```

Do we really want to track rustfmt so closely?

**Comment by djc** [comment]
No, compiling rustfmt the way CI does results in something different compared to just getting the nightly cargo-fmt.

We could potentially pin the commit for rustfmt, but so far it doesn't seem to move very fast.

**Comment by sayrer** [comment]
OK, maybe we're just in a bad spot, as they seem to have made a lot of changes in the past few months relative what ships in Rust nightly. I think an admin/fmt tool like you suggest would be a good idea, since we're operating outside of the standard toolchains.

**Comment by djc** [comment]
What ships in Rust nightly is totally different from what is used here. This is apparently switched up by the environment variables, which switch the version from 1.4.x something to 2.x something too.

**Comment by sayrer** [comment]
Yes, I understand that, it's just a barrier to contribution unless we automate it for new people.

**Comment by djc** [comment]
Sure, I agree.

**Comment by sayrer** [comment]
Addressed in #595 by @djc. Seems good to me.
